src/groupeddataframe/splitapplycombine.jl

```diff
@@ -407,31 +251,33 @@ const KWARG_PROCESSING_RULES =
     in addition to those generated. In this case if the returned
     value contains columns with the same names as the grouping columns, they are
     required to be equal.
+
+    If `regroup=false`, if the returned value should be a `DataFrame` or a
+    `GroupedDataFrame` grouped using `keycols(gdf)`.
     """
 
 """
-    combine(gd::GroupedDataFrame, args...; keepkeys::Bool=true)
-    combine(fun::Union{Function, Type}, gd::GroupedDataFrame; keepkeys::Bool=true)
-    combine(pair::Pair, gd::GroupedDataFrame; keepkeys::Bool=true)
-    combine(gd::GroupedDataFrame, fun::Union{Function, Type}; keepkeys::Bool=true)
-    combine(gd::GroupedDataFrame, pair::Pair; keepkeys::Bool=true)
+    combine(gd::GroupedDataFrame, args...; keepkeys::Bool=true, regroup::Bool=false)
+    combine(fun::Union{Function, Type}, gd::GroupedDataFrame;
+            keepkeys::Bool=true, regroup::Bool=false)
+    combine(pair::Pair, gd::GroupedDataFrame; keepkeys::Bool=true, regroup::Bool=false)
+    combine(fun::Union{Function, Type}, df::AbstractDataFrame)
+    combine(pair::Pair, df::AbstractDataFrame)
 
 Transform a [`GroupedDataFrame`](@ref) into a `DataFrame`.
 
+As a special case if `combine` is passed an `AbstractDataFrame` it applies `fun`
+or `pair` to the passed data frame as a whole.
+
 $F_ARGUMENT_RULES
 
 $F_TYPE_RULES
 
 $KWARG_PROCESSING_RULES
 
-The resulting data frame will be sorted if `sort=true` was passed to the
+The resulting data frame will be sorted by `keycols(gdf)` if `sort=true` was passed to the
 [`groupby`](@ref) call from which `gd` was constructed. Otherwise, ordering of rows
-is undefined.
-
-See also:
-- [`by(f, df, cols)`](@ref) is a shorthand for `combine(f, groupby(df, cols))`.
-- [`map`](@ref): `combine(f, groupby(df, cols))` is a more efficient equivalent
-  of `combine(map(f, groupby(df, cols)))`.
+follows the order of groups in `gdf`.
```

Comment:
So rows always appear in the order of groups in `gdf`, and when `sort=true` was passed this order reflect the sorting by `keycols(gdf)`?

Reply:
The old description was wrong, but it was late when I was writing this. The rule is simply:

> Ordering of rows follows the order of groups in `gdf`.

I have fixed this